Use SbomChecker for NTIA check for min feature

goneall opened the pull request:
~~Note: This is currently not working due to an import issue~~

goneall commented:
Here's the error I'm getting on the import:

```
spdx_dev  | Traceback (most recent call last):
spdx_dev  |   File "/spdxonlinetools/src/manage.py", line 22, in <module>
spdx_dev  |     execute_from_command_line(sys.argv)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
spdx_dev  |     utility.execute()
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/__init__.py", line 413, in execute
spdx_dev  |     self.fetch_command(subcommand).run_from_argv(self.argv)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 354, in run_from_argv
spdx_dev  |     self.execute(*args, **cmd_options)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 398, in execute
spdx_dev  |     output = self.handle(*args, **options)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 89, in wrapped
spdx_dev  |     res = handle_func(*args, **kwargs)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/commands/migrate.py", line 75, in handle
spdx_dev  |     self.check(databases=[database])
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/management/base.py", line 419, in check
spdx_dev  |     all_issues = checks.run_checks(
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/checks/registry.py", line 76, in run_checks
spdx_dev  |     new_errors = check(app_configs=app_configs, databases=databases)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/checks/urls.py", line 13, in check_url_config
spdx_dev  |     return check_resolver(resolver)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/core/checks/urls.py", line 23, in check_resolver
spdx_dev  |     return check_method()
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/urls/resolvers.py", line 416, in check
spdx_dev  |     for pattern in self.url_patterns:
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
spdx_dev  |     res = instance.__dict__[self.name] = self.func(instance)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/urls/resolvers.py", line 602, in url_patterns
spdx_dev  |     patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/utils/functional.py", line 48, in __get__
spdx_dev  |     res = instance.__dict__[self.name] = self.func(instance)
spdx_dev  |   File "/usr/local/lib/python3.10/site-packages/django/urls/resolvers.py", line 595, in urlconf_module
spdx_dev  |     return import_module(self.urlconf_name)
spdx_dev  |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
spdx_dev  |     return _bootstrap._gcd_import(name[level:], package, level)
spdx_dev  |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
spdx_dev  |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
spdx_dev  |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
spdx_dev  |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
spdx_dev  |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
spdx_dev  |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
spdx_dev  |   File "/spdxonlinetools/src/src/urls.py", line 37, in <module>
spdx_dev  |     from api import views
spdx_dev  |   File "/spdxonlinetools/src/api/views.py", line 39, in <module>
spdx_dev  |     import app.core as core
spdx_dev  |   File "/spdxonlinetools/src/app/core.py", line 19, in <module>
spdx_dev  |     from ntia_conformance_checker import SbomChecker
spdx_dev  | ImportError: cannot import name 'SbomChecker' from 'ntia_conformance_checker' (/usr/local/lib/python3.10/site-packages/ntia_conformance_checker/__init__.py)
```

goneall commented:
Note: the output has a bit of a wrapping problem and is missing newlines.

I'm also not sure if my approach to the print capture is the best for Python, but it seems to work.

Below is a screenshot of the output:
![image](https://user-images.githubusercontent.com/378172/219246843-5a54307f-6c55-48db-bff2-f5cb9a359886.png)

goneall commented:
@linynjosh - If you could do a quick review as well - thanks.

jspeed-meyers commented on src/app/core.py:
```diff
@@ -206,13 +208,21 @@ def ntia_check_helper(request):
+            """ Call the python SBOM Checker """
+            schecker = SbomChecker(str(settings.APP_DIR + uploaded_file_url))
+            oldStdout = sys.stdout
+            tempstdout = StringIO()
```
This `StringIO()` functionality could be the source of the unintended wrapping behavior. I'll investigate.

Nice work on this, PR! We'll get the output looking better next release :)

goneall commented:
It finally hit me why the formatting is off.  We're using `jQuery.text` which looses the linefeed handling.

I switched to using `jQuery.html` and added HTML line breaks in for the line feeds.

Below is the new output:
![image](https://user-images.githubusercontent.com/378172/219498206-3f51fea4-7459-4864-b8a3-cbd68d528c20.png)

goneall commented:
@jspeed-meyers @linynjosh The output could be cleaned up a bit more - we could use HTML tables, for example.  However, I think this is good enough for release - let me know if you agree.  We can clean it up a bit more in a subsequent release - perhaps with some changes to the NTIA conformance checker to make the formatting easier.

jspeed-meyers commented:
@goneall, nice debugging. I think this is "good enough" too. But for the next release, we can prioritize an improved output, perhaps in HTML if that works best.

Opened this issue: https://github.com/spdx/ntia-conformance-checker/issues/85

goneall commented:
I'll go ahead and merge this in and prep a new release.
